maven-artifact/src/main/java/org/apache/maven/artifact/versioning/DefaultArtifactVersion.java

```diff
@@ -176,6 +174,18 @@ private static Integer getNextIntegerToken(StringTokenizer tok) {
         return tryParseInt(s);
     }
 
+    private static boolean isDigits(String str) {
+        if (str == null || str.trim().isEmpty()) {
+            return false;
+        }
+        for (int i = 0; i < str.length(); i++) {
+            if (!Character.isDigit(str.charAt(i))) {
```

Comment:
This is going to allow digits from other alphabets beyond the usual 0-9. I don't think we want that, do we?